python/mlx/nn/layers/base.py

```diff
@@ -87,11 +87,77 @@ def __getattr__(self, key: str):
     def __setattr__(self, key: str, val: Any):
         self[key] = val
 
-    def load_weights(self, file: str):
+    def load_weights(
+        self, file_or_weights: Union[str, List[Tuple[str, mx.array]]], strict=True
```

Comment:
Extreme nitpick but maybe `strict: bool = True` ? I think it will feel neglected next to this huge type definition :-D .